Breeding-Insight/bi-web
# BI-1651 - Allow programs to choose/create breeding methods

# Description
**Story:** https://breedinginsight.atlassian.net/browse/BI-1651

Created a new interface under program administration to allow a breeding program to control what system breeding methods are allowed, and to create/edit/delete their own custom breeding methods.


# Dependencies
bi-api/feature/BI-1651

# Testing
1. Verify that only the `breeder` role can see the Breeding Method configuration screen
2. Verify that when first creating a program, and then navigating to the Breeding Method config screen, that all of the system breeding methods are visible
3. Verify that system breeding methods can be enabled/disabled
4. Verify that a new breeding method can be created, and that it only shows up for the program it was created in
5. Verify that germplasm can be created with program defined breeding methods
6. Verify that germplasm cannot be created with system breeding methods not enabled by the program
7. Verify that a program breeding method cannot be deleted if there is a germplasm record utilizing that method
8. Verify that a system breeding method cannot be disabled if there is a germplasm record utilizing that method


# Checklist:

- [ ] I have run TAF: https://github.com/Breeding-Insight/taf/actions/runs/3887368051
